CMakeLists.txt

```diff
@@ -222,6 +231,95 @@ endif()
 # For registration to libraw_config.h
 MACRO_BOOL_TO_01(JASPER_SUPPORT_CAN_BE_COMPILED LIBRAW_USE_REDCINECODEC)
 
+# For Adobe DNG SDK Support
+set(DNG_SDK_FOUND false)
+set(XMP_SDK_FOUND false)
+set(DNG_SDK_SUPPORT_CAN_BE_COMPILED false)
+set(XMP_CMAKE_PROJECT "${XMP_SDK_PATH}/build")
+
+if (ENABLE_DNG_SDK)
+
+    if(ZLIB_FOUND)
+
+        if(EXISTS "${DNG_SDK_PATH}/source")
+            set(DNG_SDK_FOUND true)
+        else()
+            message(STATUS "DNG SDK source code not found. Please download source code from Adobe website and set DNG_SDK_PATH")
+        endif()
+
+        if(EXISTS "${XMP_CMAKE_PROJECT}")
+            set(XMP_SDK_FOUND true)
+        else()
+            message(STATUS "XMP SDK source code not found. Please download source code from https://github.com/adobe/XMP-Toolkit-SDK and set XMP_ROOT")
+        endif()
+
+    else()
+        message(STATUS "LibZ dependency not resolved. LibRaw cannot be compiled with DNG SDK")
+    endif()
+
+    if (DNG_SDK_FOUND AND XMP_SDK_FOUND)
+        set(DNG_SDK_SUPPORT_CAN_BE_COMPILED on)
+    endif()
+endif()
+
+# For registration to libraw_config.h
+MACRO_BOOL_TO_01(DNG_SDK_SUPPORT_CAN_BE_COMPILED LIBRAW_USE_DNG_SDK)
+
+# -- Compilation rules for Adobe DNG SDK -------------------------------------------------------------
+
+if (DNG_SDK_SUPPORT_CAN_BE_COMPILED)
+
+    include_directories(${DNG_SDK_PATH}/source/)
+    include_directories(${XMP_SDK_PATH}/public/include/)
+
+    # Enable use of the DNG SDK in libraw
+    #
+    add_definitions(-DUSE_DNGSDK)
+
+    # Compile Adobe's libraries with internal symbols enabled which the DNG
+    # SDK uses.
+    #
+    add_definitions(-DAdobePrivate=1)
+
+    # Adobe's SDKs need information about the platform they are being compiled
+    # for.
+    #
+    add_definitions(-DqLinux)
+    add_definitions(-DUNIX_ENV=1)
```

Comment:
The CMake files support Windows and macOS as well. Why would this feature be restricted to Linux?

Reply:
> The CMake files support Windows and macOS as well. Why would this feature be restricted to Linux?

I just need to extend this code to detect the OS the system is building under, I'm not sure what the best way is to do this.